### DISCUSSION TITLE
Getting unexpected token import error with flow inside shared module of monorepo

### DISCUSSION THREAD

**Discussion by VivekNeel** [vercel/next.js#12914] [Help]
I have a simple project with monorepo setup. When i try to use/import a [shared module](https://github.com/VivekNeel/monorepo/blob/master/core/functions/utils.js) inside web next js [project](https://github.com/VivekNeel/monorepo/blob/master/web/pages/index.jsx#L4) am getting the below error

![image](https://user-images.githubusercontent.com/5962275/82044492-0f73db80-96cb-11ea-9312-5456a2dc65d9.png)

Please help

**Comment by Timer** [comment]
Hello! We don't run Babel against `node_modules`, so the flow types are not stripped.

You'll probably want to setup a compilation pipeline in your shared package that pre-compiles this JavaScript and checks its types.

**Reply by VivekNeel** [reply]
The fix was simple. I had to use bable.config.js instead of .babelrc which has a presets next/babel and flow stripe types.

```
module.exports = {
  presets: ['next/babel'],
  plugins: ['transform-flow-strip-types'],
};

```

**Reply by VivekNeel** [reply]
And i use next-transpile-modules to transpile my core node modules.
